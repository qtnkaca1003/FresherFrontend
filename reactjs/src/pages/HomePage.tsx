import { Button, Flex, HStack, Text } from "@chakra-ui/react"
import Layout from "components/Layout"
import { useTranslation } from "react-i18next"
import { useDispatch, useSelector } from "react-redux"
import { decrease, increase } from "redux/countSlice"
import { RootState } from "redux/store"

function HomePage() {
  const { t, i18n } = useTranslation("common")
  const count = useSelector((state: RootState) => state.count)
  const dispatch = useDispatch()
  const handleChangeJA = () => {
    i18n.changeLanguage("ja")
  }
  const handleChangeEN = () => {
    i18n.changeLanguage("en")
  }
  const handleIncrease = () => {
    dispatch(increase())
  }
  const handleDecrease = () => {
    dispatch(decrease())
  }

  return (
    <Layout
      title="Home page"
      childProps={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
      }}
    >
      <Flex
        flexDirection="column"
        justifyContent="center"
        alignItems="center"
        bg="#f2f2f2"
        p="10px"
        w="30%"
        borderRadius="20px"
      >
        <Text fontSize="2.6rem">{t("hello")}</Text>
        <HStack spacing="2rem">
          <Button colorScheme="green" fontSize="2rem" onClick={handleChangeJA}>
            Ja
          </Button>
          <Button colorScheme="blue" fontSize="2rem" onClick={handleChangeEN}>
            En
          </Button>
        </HStack>
      </Flex>
      <Flex
        flexDirection="column"
        justifyContent="center"
        alignItems="center"
        bg="#f2f2f2"
        p="2rem"
        w="30%"
        borderRadius="10px"
        mt="2rem"
      >
        <Text fontSize="2.6rem">{count.count}</Text>
        <HStack spacing="2rem">
          <Button colorScheme="red" fontSize="2rem" onClick={handleDecrease}>
            Decrease
          </Button>
          <Button colorScheme="green" fontSize="2rem" onClick={handleIncrease}>
            Increase
          </Button>
        </HStack>
      </Flex>
    </Layout>
  )
}

export default HomePage
