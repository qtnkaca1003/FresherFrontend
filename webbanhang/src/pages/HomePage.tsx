import { Text } from "@chakra-ui/react";
import axios from "axios";
import React, { useEffect, useState } from "react";
import ListProducts from "../components/home/ListProducts";
import IProduct from "../models/IProduct";
const HomePage = () => {
  const [products, setProducts] = useState<IProduct[]>([]);
  useEffect(() => {
    axios.get("https://fakestoreapi.com/products").then((res) => {
      setProducts(res.data);
    });
  },[]);
  return (
    <>
      <Text fontSize="4xl" fontWeight={"bold"} align={"center"}>
        Danh sách sản phẩm
      </Text>
      <ListProducts
        products={products}
      />
    </>
  );
};
export default HomePage;
