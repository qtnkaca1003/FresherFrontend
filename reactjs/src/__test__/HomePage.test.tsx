import HomePage from "pages/HomePage"
import { create } from "react-test-renderer"

describe("Render HomePage", () => {
  test("Matches the snapshot", () => {
    const homepage = create(<HomePage />)
    expect(homepage.toJSON()).toMatchSnapshot()
  })
})
