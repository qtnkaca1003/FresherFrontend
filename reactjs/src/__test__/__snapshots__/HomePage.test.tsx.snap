// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HomePage Matches the snapshot 1`] = `
<div
  className="css-m2yuc3"
>
  Home page
</div>
`;
